### PR TITLE
[DOCS] Reformats search API

### PR DESCRIPTION
**Pull request by szabosteve**
Relates to elastic/docs#937 and https://github.com/elastic/elasticsearch/issues/45621.

This PR updates the search API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Resources:
* [Search API spec](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/search.json)
* [Search API doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/search-search.html)

### CODE REVIEW

File: docs/reference/search/search.asciidoc [hunk 1]
```diff
@@ -1,13 +1,233 @@
...
+  be prefixed by their respective types in the response.
+  
+`version`::
+  (Optional, boolean) If `true`, returns document version as part of a hit.
+
+
+[[search-search-api-request-body]]
+==== {api-request-body-title}
+
+`description`::
```
Comment (jrodewig) [inline docs/reference/search/search.asciidoc]: This isn't explicitly included in the JSON spec, but it's valid.
```suggestion
`query`::
```

File: docs/reference/search/search.asciidoc [hunk 2]
```diff
@@ -1,13 +1,233 @@
...
+  (Optional, string) Defines the search definition using the Query DSL.
```
Comment (jrodewig) [inline docs/reference/search/search.asciidoc]:
```suggestion
  (Optional, <<query-dsl,query object>>) Defines the search definition using the <<query-dsl,Query DSL>>.
```

File: docs/reference/search/search.asciidoc [hunk 3]
```diff
@@ -1,13 +1,233 @@
...
+[[search-search-api-path-params]]
+==== {api-path-parms-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index]
+
+
+[[search-search-api-query-params]]
+==== {api-query-parms-title}
+
+`allow_no_indices`::
```
Comment (jrodewig) [inline docs/reference/search/search.asciidoc]: You can replace this def with `include::{docdir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]`

Added with #45699

File: docs/reference/search/search.asciidoc [hunk 4]
```diff
@@ -1,13 +1,233 @@
...
+  
+`df`::
+  (Optional, string) Defines the field to use as default where no field prefix 
+  is given in the query string.
+  
+`docvalue_fields`::
+  (Optional, string) A comma-separated list of fields to return as the docvalue 
+  representation of a field for each hit.
+  
+`expand_wildcards`::
```
Comment (jrodewig) [inline docs/reference/search/search.asciidoc]: You can replace this def with the following:

```
include::{docdir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
+
Defaults to `open`.
```

Will be added with #45699

File: docs/reference/search/search.asciidoc [hunk 5]
```diff
@@ -1,13 +1,233 @@
...
+  computation as part of a hit. Defaults to `false`.
+  
+`from`::
+  (Optional, integer) Defines the starting offset. Defaults to `0`.
+
+`ignore_throttled`::
+  (Optional, boolean) If `true`, concrete, expanded or aliased indices will be 
+  ignored when throttled. Defaults to `false`.
+
+`ignore_unavailable`::
```
Comment (jrodewig) [inline docs/reference/search/search.asciidoc]: Replace with `include::{docdir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]`

**Review by jrodewig — CHANGES_REQUESTED** [review]
Couple of changes. Thanks again @szabosteve.

**Comment by szabosteve** [comment]
Thanks for the review, @jrodewig. c7adf6c contains the changes you requested.

**Review by jrodewig — APPROVED** [review]
Thanks!

**Comment by szabosteve** [comment]
Backported to 7.x and 7.3.

**Pull request merged**